src/coreclr/interpreter/compiler.cpp

```diff
@@ -3530,15 +3919,84 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
                         m_ip += 5;
                         break;
                     }
+                    case CEE_ENDFILTER:
+                        AddIns(INTOP_LEAVE_FILTER);
+                        m_pStackPointer--;
+                        m_pLastNewIns->SetSVar(m_pStackPointer[0].var);
+                        m_ip++;
+                        linkBBlocks = false;
+                        break;
+                    case CEE_RETHROW:
+                        AddIns(INTOP_RETHROW);
+                        m_ip++;
+                        linkBBlocks = false;
+                        break;
                     default:
                         assert(0);
                         break;
                 }
                 break;
+
+            case CEE_ENDFINALLY:
+            {
+                AddIns(INTOP_RET_VOID);
```

Comment:
This looks weird and in the mono interpreter endfinally was its own thing. Is this correct because of funclets? I would expect the ret to match the signature of the containing method.